vmm/src/vm.rs

```diff
@@ -545,6 +535,17 @@ impl Vm {
         )
         .map_err(Error::DeviceManager)?;
 
+        let load_payload_handle = if !restoring {
+            Self::load_payload_async(
+                #[cfg(target_arch = "aarch64")]
+                &device_manager,
+                &memory_manager,
+                &config,
+            )?
+        } else {
+            None
+        };
```

Comment:
I understand for `aarch64` you need to wait for the `DeviceManager` to be created in order to provide it to `load_payload_async()`, but that's forcing you to defer the loading of the kernel to a later time, meaning you're reducing the benefit of loading the kernel asynchronously. I'll let @rbradford comment on this since he introduced the async loading, and I want to make sure this change doesn't impact too much what he did.
@jongwu is there any chance you could avoid passing the `DeviceManager` to `load_payload_async`?

Reply:
The reason to involving the `DeviceManager` is to get the `uefi_flash` device of it. While `uefi_flash` device will be created later.

Moving `uefi_flash` to `MemoryManager` can make things easy: Create `uefi_flash` when `MemoryManager` is created, then the flash is ready at this point. That way we can avoid passing `DeviceManager` here, and so avoid the `sleep` in `load_firmware()`.

Our implementation of `uefi_flash` is something vague between memory and device. Conceptually it is a device, but it can be practical to manage it in `MemoryManager`.